source/scripts/app.js

```diff
@@ -245,12 +303,41 @@ function bindExploreLoadButton() {
   let topLevel = shadow.getElementById("explore-top-level");
   let loadButton = shadow.getElementById("load-button");
 
+  let vegan = shadow.getElementById("vegan");
+  let glutenFree = shadow.getElementById("gluten-free");
+  let vegetarian = shadow.getElementById("vegetarian");
+  let input = shadow.getElementById("search-bar");
+
   loadButton.addEventListener("click", async () => {
-    if (topLevel.classList.contains("type-explore")) {
+    if (topLevel.classList.contains("type-explore") && input.value === "") {
       await populateExplorePage();
     } else {
-      // TODO: implement use case for clicking Explore Recipes when in search
-      // mode
+      if (
+        input.value === "" &&
+        !vegan.checked &&
+        !glutenFree.checked &&
+        !vegetarian.checked
+      ) {
+        topLevel.classList.toggle("type-explore");
```

Comment:
Use `toggleExplorePageType()` so the button text is also updated
```suggestion
        toggleExplorePageType();
```